src/Http/Http.Extensions/src/HeaderDictionaryTypeExtensions.cs

```diff
@@ -132,45 +132,93 @@ internal static void SetDate(this IHeaderDictionary headers, string name, DateTi
+
+    private static ContentRangeHeaderValue? ParseCacheContentRangeHeaderValue(string value) => ContentRangeHeaderValue.TryParse(value, out var result) ? result : null;
+
+    private static MediaTypeHeaderValue? ParseCacheMediaTypeHeaderValue(string value) => MediaTypeHeaderValue.TryParse(value, out var result) ? result : null;
+
+    private static RangeConditionHeaderValue? ParseCacheRangeConditionHeaderValue(string value) => RangeConditionHeaderValue.TryParse(value, out var result) ? result : null;
+
+    private static RangeHeaderValue? ParseCacheRangeHeaderValue(string value) => RangeHeaderValue.TryParse(value, out var result) ? result : null;
+
+    private static EntityTagHeaderValue? ParseCacheEntityTagHeaderValue(string value) => EntityTagHeaderValue.TryParse(value, out var result) ? result : null;
+
+    private static DateTimeOffset? ParseCacheDateTimeOffset(string value) => HeaderUtilities.TryParseDate(value, out var result) ? result : null;
+
+    private static long? ParseCacheInt64(string value) => HeaderUtilities.TryParseNonNegativeInt64(value, out var result) ? result : null;
+
+    private static IList<MediaTypeHeaderValue> ParseMediaTypeHeaderValue(IList<string> value) =>
+        MediaTypeHeaderValue.TryParseList(value, out var result) ? result : Array.Empty<MediaTypeHeaderValue>();
+
+    private static IList<StringWithQualityHeaderValue> ParseStringWithQualityHeaderValue(IList<string> value) =>
+        StringWithQualityHeaderValue.TryParseList(value, out var result) ? result : Array.Empty<StringWithQualityHeaderValue>();
+
+    private static IList<CookieHeaderValue> ParseCookieHeaderValue(IList<string> value) =>
+        CookieHeaderValue.TryParseList(value, out var result) ? result : Array.Empty<CookieHeaderValue>();
+
+    private static IList<EntityTagHeaderValue> ParseEntityTagHeaderValue(IList<string> value) =>
+        EntityTagHeaderValue.TryParseList(value, out var result) ? result : Array.Empty<EntityTagHeaderValue>();
+
+    private static IList<SetCookieHeaderValue> ParseSetCookieHeaderValue(IList<string> value) =>
+        SetCookieHeaderValue.TryParseList(value, out var result) ? result : Array.Empty<SetCookieHeaderValue>();
 
     internal static T? Get<[DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicMethods)] T>(this IHeaderDictionary headers, string name)
     {
         ArgumentNullException.ThrowIfNull(headers);
-
         var value = headers[name];
 
         if (StringValues.IsNullOrEmpty(value))
         {
             return default(T);
         }
 
-        if (KnownParsers.TryGetValue(typeof(T), out var temp))
+        object? temp = null;
+#pragma warning disable CS8974 // Converting method group to non-delegate type
+        if (typeof(T) == typeof(CacheControlHeaderValue))
+        {
+            temp = ParseCacheControlHeaderValue;
```

Comment:
Why not `return ParseCacheControlHeaderValue(value.ToString());`?  What's the benefit of wrapping it in `Func` and storing it?

Reply:
That gives an error: `CS0029	Cannot implicitly convert type 'Microsoft.Net.Http.Headers.CacheControlHeaderValue' to 'T'	Microsoft.AspNetCore.Http.Extensions`

Reply:
Personally, I'm fine with the casts, especially if it means we can get rid of the pragmas.  Obviously, confirm that it doesn't hurt perf.

Reply:
I don't see a tangible difference in performance, updated as suggested.